Jenkinsfile

```diff
@@ -1,56 +1,85 @@
-node('kafka-to-nexus') {
-    cleanWs()
-    
-    stage("Get artifacts") {
-        step([
-            $class: 'CopyArtifact',
-            filter: 'graylog-logger.tar.gz',
-            fingerprintArtifacts: true,
-            projectName: 'ess-dmsc/graylog-logger/master',
-            target: 'artifacts'
-        ])
-        dir("artifacts") {
-            sh "tar xzf graylog-logger.tar.gz"
-        }
-    }
+def project = "kafka-to-nexus"
+def centos = docker.image('essdmscdm/centos-gcc6-build-node:0.1.3')
 
-    dir("code") {
-        stage("Checkout") {
-            checkout scm
-        }
-    }
+node('docker') {
+    def container_name = "${project}-${env.BRANCH_NAME}-${env.BUILD_NUMBER}"
+    def run_args = "\
+        --name ${container_name} \
+        --tty \
+        --env http_proxy=${env.http_proxy} \
+        --env https_proxy=${env.https_proxy}"
+    def sclsh = "/usr/bin/scl enable rh-python35 devtoolset-6 -- /bin/bash"
 
-    dir("build") {
-        stage("Update local dependencies") {
-            sh "cd .. && bash code/build-script/update-local-deps.sh"
+    try {
+        container = centos.run(run_args)
+
+	cleanWs()
+
+        stage('Checkout') {
```

Comment:
We've recently changed the checkout stage for other jobs using Conan and Docker for keeping the change information on Jenkins. This can be done by cloning the repository to the node and then copying it to the container by running
```
// Outside container
dir("${project}") {
    stage('Checkout') {
        scm_vars = checkout scm
    }
}

...

// Inside container
// Copy sources to container and change owner and group.
sh "docker cp ${project} ${container_name}:/home/jenkins/${project}"
sh """docker exec --user root ${container_name} ${custom_sh} -c \"
    chown -R jenkins.jenkins /home/jenkins/${project}
\""""
```